### PR TITLE
Fix/kube reset

### PR DESCRIPTION
**Pull request by ganeshkumar1989**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #DTRSD-12328, #3016
| License          | BSD 3-Clause

## Description

An option to reset the kubeconfig has been added

### :bookmark: TODO

- [x] TRANS-27643

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                data-translate="pci_projects_project_kubernetes_service_reset_kubeconfig_description1"
+            ></p>
+            <p
+                data-translate="pci_projects_project_kubernetes_service_reset_kubeconfig_description2"
+            ></p>
+            <p
+                data-translate="pci_projects_project_kubernetes_service_reset_kubeconfig_description3"
+            ></p>
+        </div>
+        <div data-ng-if="!$ctrl.isClusterReady">
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.html:32]: You remove `div` element as `p`is a block element.

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.module.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import angular from 'angular';
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.module.js:1]:
```suggestion
import angular from 'angular';
import 'angular-translate'
import `@ovh-ux/ng-translate-async-loader`
import `ovh-ui-kit`
```
etc...

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.controller.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    this.OvhApiCloudProjectKube = OvhApiCloudProjectKube;
+  }
+
+  $onInit() {
+    this.isReseting = false;
+    this.isClusterReady = this.clusterStatus === STATUS.READY;
+  }
+
+  /**
+   * reset
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.controller.js:18]: Adding some block comments is really appreciate but is it possible to  write a
description more explicit? Or remove it if not needed at all?

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "pci_projects_project_kubernetes_service_reset_kubeconfig": "Réinitialiser le kubeconfig",
+  "pci_projects_project_kubernetes_service_reset_kubeconfig_common_confirm": "Confirmer",
+  "pci_projects_project_kubernetes_service_reset_kubeconfig_common_cancel": "Annuler",
+  "pci_projects_project_kubernetes_service_reset_kubeconfig_message": "La réinitialisation du kubeconfig entraine une courte indisponibilité de vos services !",
+  "pci_projects_project_kubernetes_service_reset_kubeconfig_description1": "Cette opération n’est pas annulable. Dès qu’elle sera lancée, vous perdrez l’accès à votre API Server avec votre kubeconfig actuel. Les nœuds perdront également accès à l’API server mais les conteneurs hébergés continueront à fonctionner.",
+  "pci_projects_project_kubernetes_service_reset_kubeconfig_description2": "Vous devrez télécharger un nouveau fichier kubeconfig quelques minutes plus tard (temps de la génération de la nouvelle autorité de certification et mise à jour des composants du control plane avec cette nouvelle CA). Chaque nœud sera ensuite réinstallé (l’un après l’autre) pour retrouvera l’accès à l’APIserver à l’issue de cette dernière étape.",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/translations/Messages_fr_FR.json:7]: ⚠️ Some typos

```suggestion
  "pci_projects_project_kubernetes_service_reset_kubeconfig_description2": "Vous devrez télécharger un nouveau fichier kubeconfig quelques minutes plus tard (temps de la génération de la nouvelle autorité de certification et mise à jour des composants du control panel avec cette nouvelle CA). Chaque nœud sera ensuite réinstallé (l’un après l’autre) pour retrouver l’accès à l’API server à l’issue de cette dernière étape.",
```

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.routing.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        clusterStatus: null,
+      },
+      views: {
+        modal: {
+          component: 'pciProjectKubernetesServiceResetKubeconfig',
+        },
+      },
+      layout: 'modal',
+      resolve: {
+        clusterStatus: /* @ngInject */ ($transition$) => {
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.routing.js:16]: Isn't `clusterStatus` inherited from parent state within a `cluster` resolve or something similar ?

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.html (hunk 1 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.html]: This is done

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.module.js (hunk 2 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.module.js:1]: This is fixed :)

**Comment by ganeshkumar1989** [comment]
> According to the `ovh-api-services` changelog, you've updated the `$resource`
> by adding a new (missing) action which is more like a `feature`.
> 
> See: [ovh-ux/ovh-api-services@66d04d0](https://github.com/ovh-ux/ovh-api-services/commit/66d04d08aba4580b54cd5e979ad53fb094e12490)
> 
> Updating this dependency across all packages w/ the Monorepo it's not like
> **fixing** something but more like updating/bumping a dependency.
> 
> So I encourage you to rename the following commit `a73253751784061533105fa6ac38510df068f7a7`
> by using the `build` commit type like:
> 
> ```
> build(deps): upgrade ovh-api-services to v9.44.1
> ```
> 
> Is that make sense for you?
> 
> Thanks!

Got it. Thanks for the explanation @antleblanc . The commit has been reworded as suggested.

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.controller.js (hunk 3 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.controller.js]: Agreed. The piece of code was copied from another place which had this comment. It has been removed now. Thanks

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.routing.js (hunk 5 above)
Comment (ganeshkumar1989) [reply packages/manager/modules/pci/src/projects/project/kubernetes/details/service/reset-kubeconfig/reset-kubeconfig.routing.js]: The cluster object was being passed from the parent. The status is not being retrieved from the cluster object instead. Thanks

**Comment by ganeshkumar1989** [comment]
Translations complete and ready for prod

**Review by antleblanc — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**